plugin/etcd/msg/type.go

```diff
@@ -18,6 +19,10 @@ func (s *Service) HostType() (what uint16, normalized net.IP) {
 
 	ip := net.ParseIP(s.Host)
 
+	if s.Key != "" && strings.HasPrefix(Domain(s.Key), "_") {
```

Comment:
Why do we need this?
HostType() can't return TypeSRV as descibed in the function docs.

Reply:
This is used in the backend_lookup.go AXFR() so we can transfer the SRV records. Otherwise, we're limited to transferring only the types that HostType() supports ( A/AAAA/CNAME ). I can update the comment to show that it's supported.